src/NetscriptFunctions.ts

```diff
@@ -288,43 +288,45 @@ export const ns: InternalAPI<NSFull> = {
     },
   growthAnalyze:
     (ctx) =>
-    (_hostname, _growth, _cores = 1) => {
-      const hostname = helpers.string(ctx, "hostname", _hostname);
-      const growth = helpers.number(ctx, "growth", _growth);
+    (_host, _multiplier, _cores = 1) => {
+      const host = helpers.string(ctx, "hostname", _host);
+      const mult = helpers.number(ctx, "multiplier", _multiplier);
       const cores = helpers.number(ctx, "cores", _cores);
 
       // Check argument validity
-      const server = helpers.getServer(ctx, hostname);
+      const server = helpers.getServer(ctx, host);
       if (!(server instanceof Server)) {
-        helpers.log(ctx, () => "Cannot be executed on this server.");
+        // Todo 2.3: Make this throw instead of returning 0?
+        helpers.log(ctx, () => `${host} is not a hackable server. Returning 0.`);
         return 0;
       }
-      if (typeof growth !== "number" || isNaN(growth) || growth < 1 || !isFinite(growth)) {
-        throw helpers.makeRuntimeErrorMsg(ctx, `Invalid argument: growth must be numeric and >= 1, is ${growth}.`);
+      if (mult < 1 || !isFinite(mult)) {
+        throw helpers.makeRuntimeErrorMsg(ctx, `Invalid argument: multiplier must be finite and >= 1, is ${mult}.`);
+      }
+      // TODO 2.3: Add assertion function for positive integer, there are a lot of places everywhere that can use this
+      if (cores < 1 || !isFinite(cores)) {
+        throw helpers.makeRuntimeErrorMsg(ctx, `Invalid argument: cores must be finite and >= 1, is ${cores}.`);
       }
 
-      return numCycleForGrowth(server, Number(growth), cores);
+      return Math.ceil(numCycleForGrowth(server, mult, cores));
```

Comment:
Please **don't** do this.

Not only is it a breaking change, which we'd like to avoid, but it destroys much of the utility that growthAnalyze has over growthAnalyzeCorrected. Despite the warning, most users actually *want* the full precision, and probably would prefer it if growthAnalyzeCorrected gave decimal results too. (But it can't, not at full accuracy, without being needlessly slower.)

For example, a common recommendation is to hardcode "Math.E" as the value to pass for `mult`, since this makes future math simpler. If you do this for n00dles, you get a pretty small number, well less than 50. Truncating this to an integer would kill this method; you'd have to pass a a value near infinity just to get acceptable precision back.